UpgradedVehicles/VehicleUpgrader.cs

```diff
@@ -89,21 +294,53 @@ private float GetSpeedMultiplierBonus(float speedBoosterCount)
         {
             float speedBoosterRatio = 0f;
             float bonusSpeedRatio = 0f;
+#if BELOWZERO
+            float SennaSpeedMultiplier = 1f;
+#endif
 
-            if (IsExosuit)
+            switch (currentVehicleType)
             {
-                bonusSpeedRatio = _exosuitBonusSpeedMultiplier;
-                speedBoosterRatio = 0.45f;
+                case EVehicleType.Exosuit:
+                    bonusSpeedRatio = _exosuitBonusSpeedMultiplier;
+                    speedBoosterRatio = 0.45f;
+                    break;
+
+                case EVehicleType.Seamoth:
+                    bonusSpeedRatio = _seamothBonusSpeedMultiplier;
+                    speedBoosterRatio = 0.35f;
+                    break;
+
+#if BELOWZERO
+                case EVehicleType.Seatruck:
+                    bonusSpeedRatio = _seatruckBonusSpeedMultiplier;
+                    speedBoosterRatio = 0.35f;
+                    break;
+#endif
             }
 
-            if (IsSeamoth)
+#if SUBNAUTICA
+            return 1f + // Base 100%
```

Comment:
Potential syntax error, trailing operator
```cs
 return 1f +
```

Reply:
It's fine. Look at the whole block: if SUBNAUTICA is defined, the entire block resolves as 
```
            return 1f + // Base 100%
            speedBoosterCount * speedBoosterRatio + // Bonus from Speed Boosters
                   this.DepthIndex * bonusSpeedRatio; // Bonus from Depth Modules

```
If BELOWZERO is defined instead, it resolves as
```
            if (SeaTruckSpeedManager != null)
            {
                SennaSpeedMultiplier = 1f;

                foreach (KeyValuePair<TechType, float> armorModule in SpeedManagerAccelerations)
                {
                    TechType techType = armorModule.Key;
                    if (this.UpgradeModules.GetCount(techType) > 0)
                    {
                        float multiplier = armorModule.Value;
                        SennaSpeedMultiplier = Math.Max(SennaSpeedMultiplier, multiplier);
                    }
                }

                //ErrorMessage.AddMessage($"GetImpactArmorFraction: armorModuleCount = {armorModuleCount}");
            }

            return SennaSpeedMultiplier +
            speedBoosterCount * speedBoosterRatio + // Bonus from Speed Boosters
                   this.DepthIndex * bonusSpeedRatio; // Bonus from Depth Modules

```